llvm/lib/Target/TVM/TVMStackFixup.cpp

```diff
@@ -805,7 +805,9 @@ operator()(const std::pair<Change, std::string> &pair) const {
 
   assert(MI && "MI is not generated");
 
-  MFI->addStackModelComment(MI, pair.second);
+  if constexpr (PrintStackInAsm) {
+    MFI->addStackModelComment(MI, pair.second);
```

Comment:
Just for my curiosity: can you elaborate why this is a problem?

Reply:
If it is enabled, it produces a lot of excessive data(stack values) in asm file, like this:
```
PUSHINT	21
ROLLX         ; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | %35 | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | %32 | %76 | x | x | x | x | %433 | x | %437 | %60 | - }
SWAP          ; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | %35 | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | %32 | %76 | x | x | x | x | %433 | x | %60 | %437 | - }
STU	32      ; >%438 = STU %60, %437, i257 32
		; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | %35 | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | %32 | %76 | x | x | x | x | %433 | x | %438 | - }
PUSHINT	30
ROLLX         ; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | %32 | %76 | x | x | x | x | %433 | x | %438 | %35 | - }
SWAP          ; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | %32 | %76 | x | x | x | x | %433 | x | %35 | %438 | - }
STREF         ; >%439 = STREF %35, %438
		; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | %32 | %76 | x | x | x | x | %433 | x | %439 | - }
XCHG	s1, s8  ; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | x | %76 | x | x | x | x | %433 | %32 | %439 | - }
STU	1       ; >%440 = STU %32, %439, i257 1
		; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | x | %76 | x | x | x | x | %433 | %440 | - }
PUSH	s34     ; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | x | %76 | x | x | x | x | %433 | %440 | %2 | - }
SWAP          ; { x | %1405 | x | %1436 | %4 | %5 | %6 | x | %1392 | %2 | x | %29 | x | %1393 | x | x | x | x | x | x | %49 | x | x | x | x | %1389 | x | x | %1413 | %1418 | %61 | x | x | x | x | x | x | %76 | x | x | x | x | %433 | %2 | %440 | - }
```
And when `PrintStackInAsm` is false:
```
PUSHINT	21
ROLLX
SWAP
STU	32      ; >%438 = STU %60, %437, i257 32
PUSHINT	30
ROLLX
SWAP
STREF         ; >%439 = STREF %35, %438
XCHG	s1, s8
STU	1       ; >%440 = STU %32, %439, i257 1
PUSH	s34
SWAP
```
This data is helpful when you debug compiler itself, e.g. some stack optimizations, but useless and too verbose when you're just trying to debug compiled code itself. For example, it is hard to search def-use with this data.